internal/controller/storageclient_controller.go

```diff
@@ -606,7 +608,15 @@ func (r *storageClientReconcile) newExternalClusterClient() (*providerClient.OCS
 func (r *storageClientReconcile) onboardConsumer(externalClusterClient *providerClient.OCSProviderClient, operatorVersion string) error {
 	onboardRequest := providerClient.NewOnboardConsumerRequest().
 		SetOnboardingTicket(r.storageClient.Spec.OnboardingTicket).
-		SetClientOperatorVersion(operatorVersion)
+		SetClientOperatorVersion(operatorVersion).
+		SetClientOperatorNamespace(r.OperatorNamespace).
+		SetClientName(r.storageClient.Name).
+		SetClientID(string(r.storageClient.UID)).(interfaces.StorageClientOnboarding)
```

Comment:
This is a bad situation to use chaining because the second op changes the returned type. This is also why you need to have a, interface type cast at the end
```suggestion
	onboardRequest := providerClient.NewOnboardConsumerRequest()
	onboardRequest.SetOnboardingTicket(r.storageClient.Spec.OnboardingTicket)
	onboardRequest.SetClientOperatorVersion(operatorVersion)
	onboardRequest.SetClientOperatorNamespace(r.OperatorNamespace)
	onboardRequest.SetClientName(r.storageClient.Name)
	onboardRequest.SetClientID(string(r.storageClient.UID))
```